jupyterhub/spawner.py

```diff
@@ -204,12 +199,19 @@ def clear_state(self):
         self.api_token = ''
     
     def get_env(self):
-        """Return the environment we should use
-        
-        Default returns a copy of self.env.
+        """Return the environment dict to use for the Spawner.
+
+        This applies things like `env_keep`, anything defined in `Spawner.env`,
+        and adds the API token to the env.
+
         Use this to access the env in Spawner.start to allow extension in subclasses.
         """
-        return self.env.copy()
+        env = self.env.copy()
+        for key in self.env_keep:
```

Comment:
I think reversing the priority should be fine - env_keep lowest, self.env highest, since this change makes `self.env` clearly and explicitly env-overrides.